packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -189,6 +190,37 @@ export class ChromeService {
     this.initVisibility(application);
     this.handleEuiFullScreenChanges();
 
+    // Ensure developers are notified if working in a context that lacks the EUI Provider.
+    const isDev = this.params.coreContext.env.mode.name === 'development';
+    if (isDev) {
```

Comment:
I know we haven't done it much in this `start` method but any chance we would not keep on adding logic and make it grow any further by moving this logic into its own function/class method?